[JIT][torchbind] Namespaces for torchbind classes

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35054 [JIT][torchbind] Namespaces for torchbind classes**

This changes it so that we have namespaces on class names. This makes us more in line with ops (e.g. ops.foo.bar) and sets us up for the loading system we've been sketching out

Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

Commit:
[JIT][torchbind] Namespaces for torchbind classes

[ghstack-poisoned]

zdevito commented on torch/custom_class.h:
```diff
@@ -55,8 +55,8 @@ class class_ {
   /// `torch.classes.MyStack` in both Python and TorchScript.
-  explicit class_(const std::string& className) : className(std::move(className)) {
-    qualClassName = topModule + "." + parentModule + "." + className;
+  explicit class_(const std::string& namespaceName, const std::string& className) {
```
this needs to sanitize namespaceName/className or users can get really weird errors.

dzhulgakov replied:
random idea: shall we support "namespace::foo" syntax similarly to op definition API?

and yea, error out if name contains '.' or '::' is good to prevent mistakes

jamesr66a replied:
@dzhulgakov that's an interesting point: I would so for now no because we have two disjoint naming conventions in the system:

* the op namespace thing (e.g. 'aten::foo'`
* QualifiedName (foo.bar.baz.Quux)

It may be something that we want to bring up with the op registration effort. Can we reconcile these things?

zdevito reviewed (approved):
Code looks good. Did anyone serialize models with the old xnnpack operators in it? If so there needs to be backward compat code added.

Commit:
Update on "[JIT][torchbind] Namespaces for torchbind classes"


Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

[ghstack-poisoned]

Commit:
Update on "[JIT][torchbind] Namespaces for torchbind classes"


Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

[ghstack-poisoned]

Commit:
Update on "[JIT][torchbind] Namespaces for torchbind classes"


Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

[ghstack-poisoned]

Commit:
Update on "[JIT][torchbind] Namespaces for torchbind classes"


Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

[ghstack-poisoned]

Commit:
Update on "[JIT][torchbind] Namespaces for torchbind classes"


Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

[ghstack-poisoned]

Commit:
Update on "[JIT][torchbind] Namespaces for torchbind classes"


This changes it so that we have namespaces on class names. This makes us more in line with ops (e.g. ops.foo.bar) and sets us up for the loading system we've been sketching out

Differential Revision: [D20541090](https://our.internmc.facebook.com/intern/diff/D20541090)

[ghstack-poisoned]
